src/map/battle.cpp

```diff
@@ -2617,7 +2632,7 @@ static bool is_attack_critical(struct Damage* wd, struct block_list *src, struct
 			case SN_SHARPSHOOTING:
 			case MA_SHARPSHOOTING:
 #ifdef RENEWAL
-				cri += 300; // !TODO: Confirm new bonus
+				cri += cri / 2;
```

Comment:
It said Critical modifier from itembonus half not critical rate.

Reply:
Sigma has it quoted as `-The half of user's Cri chance is added to the skill base cri chance.`. Did I interpret this wrong?

Reply:
```
#ifdef RENEWAL
	if (is_attack_critical(&wd, src, target, skill_id, skill_lv, false)) {
		if (sd) { //Check for player so we don't crash out, monsters don't have bonus crit rates [helvetica]
			wd.damage = (int64)floor((float)((wd.damage * 140) / 100 * (100 + sd->bonus.crit_atk_rate/2)) / 100); //Crit Stuff Here
			if (is_attack_left_handed(src, skill_id))
				wd.damage2 = (int64)floor((float)((wd.damage2 * 140) / 100 * (100 + sd->bonus.crit_atk_rate/2)) / 100); //Crit Stuff Here
		} else
			wd.damage = (int64)floor((float)(wd.damage * 140) / 100);

		if (tsd && tsd->bonus.crit_def_rate != 0) {
			ATK_ADDRATE(wd.damage, wd.damage2, -tsd->bonus.crit_def_rate);
		}
	}
```

Reply:
Crit attack rate should be halven on only skill damage, not on normal attack damage. How about this? I have tested it and I think it worked.

> if (is_attack_critical(wd, src, target, skill_id, skill_lv, false)) {
		if (sd) { //Check for player so we don't crash out, monsters don't have bonus crit rates [helvetica]
			if(skill_id > 0)
				wd.damage = (int)floor((float)((wd.damage * 140) / 100 * (100 + (sd->bonus.crit_atk_rate/2))) / 100);
			else
				wd.damage = (int)floor((float)((wd.damage * 140) / 100 * (100 + sd->bonus.crit_atk_rate)) / 100);
			if (is_attack_left_handed(src, skill_id))
				if(skill_id > 0)
					wd.damage2 = (int)floor((float)((wd.damage2 * 140) / 100 * (100 + (sd->bonus.crit_atk_rate/2))) / 100);
				else
					wd.damage2 = (int)floor((float)((wd.damage2 * 140) / 100 * (100 + sd->bonus.crit_atk_rate)) / 100);
		} else
			wd.damage = (int)floor((float)(wd.damage * 140) / 100);